lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -1903,18 +1903,21 @@ public Object readSqlResultRecord(final ResultSet result) {
 	}
 
 	public List<SimplePage> getTopLevelPages(final String siteId) {
-		DetachedCriteria d = DetachedCriteria.forClass(SimplePage.class).add(Restrictions.eq("siteId", siteId))
-			.add(Restrictions.disjunction()
-				.add(Restrictions.isNull("owner"))
-				.add(Restrictions.eq("owned", true)))
-			.add(Restrictions.isNull("parent"));
+	    // set of all top level pages, actually the items pointing to them                                                                       
+	    List<SimplePageItem> sitePages =  findItemsInSite(toolManager.getCurrentPlacement().getContext());
```

Comment:
Thanks for fixing things up @clhedrick.

My only query is regarding the use of `findItemsInSite`. Performance is important for this feature, as we try to make the subnav initialisation relatively seamless upon page load.  Does the change to use this increase the performance overhead enough for us to worry?

Reply:
We should worry about this change! This is going to change this lookup from one query into possibly hundreds or thousands of queries (site => pages => items), right?!